### PR TITLE
feat(telemetry): Include user intent for copy and insert code interactions

### PR DESCRIPTION
**Pull request by lalexcaws**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

Adding User Intent to copy code and insert code at cursor interaction telemetry data
[Why] This will be part of an effort to capture metrics on how often each interaction type is occurring when the user intent is related to unit test generation

## Checklist
- [X] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rli** [comment]
```
> Task :plugin-amazonq:chat:jetbrains-community:compileKotlin
e: file:///codebuild/output/src152308433/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt:213:21 Cannot find a parameter with this name: cwsprChatUserIntent
e: file:///codebuild/output/src152308433/src/github.com/aws/aws-toolkit-jetbrains/plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt:237:21 Cannot find a parameter with this name: cwsprChatUserIntent
Finished executing kotlin compiler using DAEMON strategy

> Task :plugin-amazonq:chat:jetbrains-community:compileKotlin FAILED

```

File: plugins/core/jetbrains-community/resources/telemetryOverride.json [hunk 1]
```diff
@@ -653,6 +653,10 @@
                 {
                     "type": "cwsprChatMessageId"
                 },
+                {
+                    "type": "cwsprChatUserIntent",
+                    "required": false
+                },
```
Comment (rli) [inline plugins/core/jetbrains-community/resources/telemetryOverride.json:659]: should be pushing this into the common definitions

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/resources/telemetryOverride.json (hunk 1 above)
Comment (lalexcaws) [reply plugins/core/jetbrains-community/resources/telemetryOverride.json:659]: yep, this was mentioned in my VSCode plugin change as well. Going to get out that change

**Review by laileni-aws — APPROVED** [review]

**Comment by lalexcaws** [comment]
Commons definition update: https://github.com/aws/aws-toolkit-common/pull/864

**Comment by rli** [comment]
can you do a follow-up to remove the override and use the common definition?

**Pull request merged**